src/app/options/options.component.spec.ts

```diff
@@ -18,10 +21,51 @@ describe('OptionsComponent', () => {
   beforeEach(() => {
     fixture = TestBed.createComponent(OptionsComponent);
     component = fixture.componentInstance;
+    component.formats = [ "test1", "test2 [Read-only]", "test3 [Write-only]" ];
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should filter out write-only results from input drop-down list', async () => {
+    await fixture.whenStable();
+    const input = fixture.debugElement.query(By.css('.inputAutocomplete')).nativeElement;
+
+    expect(input.innerText).toEqual(" test1 test2 [Read-only]");
+  });
+
+  it('should filter results for input drowdown list based on input', async () => {
+    await fixture.whenStable();
+    const inputInput = fixture.debugElement.query(By.css('.inputInput')).nativeElement;
+
+    inputInput.value = "1";
+    inputInput.dispatchEvent(new Event('input'));
+    fixture.detectChanges();
+    const input = fixture.debugElement.query(By.css('.inputAutocomplete')).nativeElement;
+
+    expect(input.innerText).toEqual(" test1");
+  });
+
+  it('should filter out read-only results from output drop-down list', async () => {
+    await fixture.whenStable();
+    const output = fixture.debugElement.query(By.css('.outputAutocomplete')).nativeElement;
+
+    expect(output.innerText).toEqual(" test1 test3 [Write-only]");
+  });
+
+  it('should filter results for output drowdown list based on input', async () => {
```

Comment:
Same as note above. Use MatInputHarness or MatAutocompleteHarness to simulate actual clicks done by users. 

https://material.angular.io/guide/using-component-harnesses#using-a-harness-to-interact-with-an-angular-material-component

https://material.angular.io/components/input/api#classes

Example code:

import {MatAutocompleteHarness} from '@angular/material/autocomplete/testing';

const autocompleteHarness = await getHarness(MatAutocompleteHarness);
await autocompleteHarness.focus();
await autocompleteHarness.enterText('test');
const options = await autocompleteHarness.getOptions();
expect(options) ..toEqual(['test1, ....'])